service/src/main/java/tds/assessment/repositories/impl/ItemQueryRepositoryImpl.java

```diff
@@ -0,0 +1,133 @@
+                        "    A.groupkey,\n" +
+                        "    A.itemposition AS position,\n" +
+                        "    A.isactive,\n" +
+                        "    A.isfieldtest,\n" +
+                        "    A.isrequired, \n" +
+                        "    A.strandname \n" +
+                        "FROM \n" +
+                        "    itembank.tblsetofadminitems as A \n" +
+                        "JOIN \n" +
+                        "    itembank.tblitem I \n" +
+                        "    ON I._key = A._fk_item \n" +
+                        "JOIN itembank.tblsetofadminsubjects segments \n" +
+                        "    ON segments._key = A._fk_adminsubject \n" +
+                        "WHERE \n" +
+                        "   segments.virtualtest = :key OR segments._key = :key";
+
+        List<Item> items = jdbcTemplate.query(itemsSQL, parameters, (RowMapper<Item>) (rs, row) -> {
+                    Item item = new Item(rs.getString("id"));
+                    item.setSegmentKey(rs.getString("segmentKey"));
+                    item.setItemType(rs.getString("itemtype"));
+                    item.setGroupId(rs.getString("groupid"));
+                    item.setGroupKey(rs.getString("groupkey"));
+                    item.setPosition(rs.getInt("position"));
+                    item.setActive(rs.getBoolean("isactive"));
+                    item.setFieldTest(rs.getBoolean("isfieldtest"));
+                    item.setRequired(rs.getBoolean("isrequired"));
+                    item.setStrand(rs.getString("strandname"));
+                    return item;
+                }
+        );
+
+
+
+        return items;
+    }
+
+    @Override
+    public List<ItemProperty> findActiveItemsProperties(final String assessmentKey) {
+        final SqlParameterSource parameters = new MapSqlParameterSource("key", assessmentKey);
+
+        final String SQL =
+            "SELECT \n" +
+            "   _fk_item AS itemId, \n" +
+            "   propname AS name, \n" +
+            "   propvalue AS value, \n" +
+            "   propdescription AS description \n" +
+            "FROM \n" +
+            "   itembank.tblitemprops P \n" +
+            "JOIN itembank.tblsetofadminsubjects segments \n" +
+            "    ON segments._key = P._fk_adminsubject \n" +
+            "WHERE \n" +
+            "   segments.virtualtest = :key OR segments._key = :key AND" +
+            "   isActive = 1";
+
+        return jdbcTemplate.query(SQL, parameters, (rs, row) ->
+            new ItemProperty(
+                rs.getString("name"),
+                rs.getString("value"),
+                rs.getString("description"),
+                rs.getString("itemId")
+            )
+        );
+    }
+
+    @Override
+    public List<ItemConstraint> findItemConstraintsForAssessment(final String assessmentId) {
+        Map<String, Object> parameters = new HashMap<>();
+//        parameters.put("clientName", clientName);
```

Comment:
I would like to include it in this pull request. I wanted to get some feedback (as discussed at standup) on whether or not we should include the clientName in the GET Assessment API url params. I've been leaning towards "yes".